src/Stoa.ts

```diff
@@ -1462,6 +1480,9 @@ class Stoa extends WebService
                         {
                             if (JSBI.equal(block.header.height.value, expected_height.value))
                             {
+                                for (let tx of block.txs)
+                                    await this.transaction_pool.remove(tx);
```

Comment:
You can move the call to the point. `ledger_storage.putBlocks `

Reply:
Yeah, I was thinking about that, too. Now that we've found a good way, we'd better do it.